backend/movement/movement.py

```diff
@@ -0,0 +1,249 @@
+            current, path = queue.pop(0)
+            if current in destinations:
+                return path
+            if current[0] < 0 or current[0] >= width or current[1] < 0 or current[1] >= height:
+                continue
+            if current in obstacle_locations:
+                continue
+            if current in visited:
+                continue
+            visited.add(current)
+            for i, j in [(0, 1), (0, -1), (1, 0), (-1, 0)]:
+                next_pos = (current[0] + i, current[1] + j)
+                new_path = path.copy()
+                new_path.append(next_pos)
+                queue.append((next_pos, new_path))
+        assert False, "No path found"
+    
+    def move(self, state, player, destination, action, param_arg_dict):
+        """
+        Moves the player to the destination.
+
+        If animate mode is disabled, the player is immediately moved to the
+        destination. If animate mode is enabled, the player path is calculated
+        and the player moves the first step in the path.
+
+        Args:
+            state (State): The state of the environment.
+            player (Object): The player to move.
+            destination (Object): The destination station.
+            action (Action): The move action.
+            param_arg_dict (Dictionary[str, Object]): The arguments of the action.
+
+        Modifies:
+            player (Player): Modifies the path and pos fields of the player.
+        """
+        player_obj = self.players[player.name]
+        destination_pos = self.stations[destination.name]
+        possible_destinations = self._get_possible_destinations(player_obj, destination_pos)
+        # If player is already at the destination, perform move action immediately
+        if player_obj.pos in possible_destinations:
+            action.perform_action(state, param_arg_dict)
+            player_obj.direction = (destination_pos[0] - player_obj.pos[0], destination_pos[1] - player_obj.pos[1])
+            return
+        # Get the path to the destination
+        path = self._get_player_path(player_obj, possible_destinations)
+        # If animate mode is disabled, move player to destination immediately
+        if not self.animate:
+            player_obj.pos = path[-1]
+            player_obj.direction = (destination_pos[0] - player_obj.pos[0], destination_pos[1] - player_obj.pos[1])
+            action.perform_action(state, param_arg_dict)
+        else:
+            # Animate the movement by moving the player by on step in the path
+            player_obj.path = path
+            next_pos = player_obj.path.pop(0)
+            player_obj.direction = (next_pos[0] - player_obj.pos[0], next_pos[1] - player_obj.pos[1])
+            player_obj.pos = next_pos
+            # If player has reached the destination, perform the action
+            if player_obj.path == []:
+                action.perform_action(state, param_arg_dict)
+                destination = param_arg_dict["s2"]
```

Comment:
is it possible to combine this with the logic before returning if the player is already at the destination?